tests/e2escenarios/e2e_devfile_test.go

```diff
@@ -0,0 +1,174 @@
+package e2escenarios
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// Test Objective:
+//    Test ODO devfile support features
+
+// Scope:
+//    Test debug support for the following components, making use of starter projects define in the corresponding devfile:
+//    - nodejs
+//    - java-springboot
+//    - java-quarkus
+//    - java-openliberty
+//    - java-maven, no starter project available at the time this script was developed, so skiping this component for now
+
+var _ = Describe("odo devfile supported tests", func() {
+	var componentName, projectDirPath string
+	var projectDir = "/projectDir"
+	var commonVar helper.CommonVar
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		componentName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+		projectDirPath = commonVar.Context + projectDir
+	})
+
+	preSetup := func() {
```

Comment:
IMO we should not have another function layer as `preSetup` because this preSetup has been called globally in the script. So, I think we can include these two steps in `BeforeEach` itself. Correct me if I am wrong.

Reply:
Addressed in commit "#3377 addressed all Priti's comments in PR"